engine/src/main/java/org/destinationsol/game/Hero.java

```diff
@@ -89,6 +97,58 @@ public void setSolShip(SolShip hero, SolGame solGame) {
         solGame.getSolApplication().getMusicManager().playMusic(OggMusicManager.GAME_MUSIC_SET, options);
     }
 
+    public void changeShip(Hero hero, String newShipID, SolGame game, Console console) {
+        SolShip originalShip = hero.getShip();
+        Optional<SolShip> newShip;
+
+        boolean isARealShip = false;
+
+        for (ResourceUrn urn : Assets.getAssetHelper().list(Json.class)) {
+            if ((urn.getModuleName() + ":" + urn.getResourceName()).equals(newShipID)) {
+                isARealShip = true;
```

Comment:
This here is wrong. What it's really doing is checking whether the given string exists as a Gestalt asset. When it's anything but a ship it causes issues down the line. I know it wasn't your code, just pointing out the issue. :+1: